### PR TITLE
Prevent model from accidentally freezing its own properties

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @MariagraziaAlastra 

## Description:

In https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/153 we noticed that the model's own properties were accidentally getting frozen (by [this line](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/7d0afb1dbdd0430e30d92f1228d654d28364c7d4/shared/js/ui/base/store.es6.js#L60)) because they were being passed by reference.

This PR makes sure anything passed to the store is deep cloned so that problem doesn't happen. It also removes a hack we had to add in the whitelist model.

## Steps to test this PR:

1. Try adding and removing some items in the whitelist - it should work without any errors
2. Other models should also work fine.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/ui/base/model.es6.js [hunk 1]
```diff
@@ -57,6 +57,9 @@ BaseModel.prototype = $.extend({},
       const lastValue = this[attr] || null
       this[attr] = val
 
+      // deep clone val in case it's something passed by reference
+      val = JSON.parse(JSON.stringify(val))
```
Comment (andrey-p) [inline shared/js/ui/base/model.es6.js]: I was considering doing this to `lastValue` as well. Not sure if it's worth the extra processing time for it, through.

Comment (MariagraziaAlastra) [reply shared/js/ui/base/model.es6.js]: I think it's fine, lastValue seems something we wouldn't modify after sending a notification

Comment (MariagraziaAlastra) [reply shared/js/ui/base/model.es6.js]: @andrey-p would you mind checking whether JSON.stringify(val) is defined though? If val is null, that method returns undefined, and JSON.parse() will throw a SyntaxError.

Comment (andrey-p) [reply shared/js/ui/base/model.es6.js]: Good find, lemme fix!

**Pull request merged**